### DISCUSSION TITLE
Dependent dlls for Migrator.exe?

### DISCUSSION THREAD

**Discussion by GKannan1990** [fluentmigrator/fluentmigrator#1613] [Q&A] [answered]
Hi Team,

I'm using Fluent Migrator Console 3.3.2 in my.Net application. 

To migrate up or down the database, Migrate.exe needs to be executed by passing the assembly file from the command line. And by default, the Migrate.exe file resides in the /packages/FluentMigrator.Console.3.3.2/net461/any folder.

I want to bundle this Migrator.exe with my setup. I removed other database dlls. However, I am receiving an error.

Could you please let me know the dependent dlls for this Migrator.exe? so that I can bundle those dlls. 

We are using an SQLite database.

**Comment by jzabroski** [comment, marked as answer]
Use FluentMigrator.DotNet.Cli and install it as a .NET local tool

**Reply by GKannan1990** [reply]
Thanks for quick reply. We tried  FluentMigrator.DotNet.Cli, it only works for .Net Core.  
We are working in .Net framework 4.8 and with Migrator.exe we need to include all the dlls which is increasing our installer size due to unwanted dlls. 

Can you please suggest

**Reply by jzabroski** [reply]
What you were proposing is illogical and dangerous. You don't just copy parts of an executable and try to run it in a different directory -- for _any_ application. What made you think this was a reasonable approach that wouldn't end up with a major issue at some point?

We have had [discussions previously](https://github.com/fluentmigrator/fluentmigrator/issues/812#issuecomment-1134825126) where you were doing things highly questionable, like using a 22 year old version of SQLite (1.0.*) https://www.sqlite.org/changes.html 

I don't know if you're trying to do this because someone managing you is suggesting it, but if they are, they're just extremely willing to take risks I would never think of taking.

> Can you please suggest

Upgrade from .NET Framework 4.8 to .NET Core for your migrations.  This means you likely can't directly access your Data Access Library project from your migrations DLL.  Decide if that is a deal breaker for you.  If it isn't, then it's time to reason with your manager about his or her approach to solving problems.  If he or she isn't willing to start moving off .NET Framework 4.8 even for a fairly isolated dependency like migrations, you'll probably be stuck maintaining .NET Framework code forever.  I realize your ultimate goal is to decrease your installer size, but that **should not be your goal**, since it's a terrible business strategy (how will you recruit talent to work on .NET Framework?). If it actually is your team's goal, this project doesn't share that goal for .NET Framework. This project's goal for .NET Framework is to keep the lights on for customers stuck on .NET Framework, and **that's it.** .NET Framework is in maintenance mode and Microsoft isn't shipping features for it _and hasn't for years_, so there should be no expectation any open source project would ship new features, either.
